### PR TITLE
use raw strings to avoid escaping

### PR DESCRIPTION
**Pull request by a14n**
## Description

A raw string can be used to avoid escaping only backslashes and dollars.

**BAD:**
```dart
var s = 'A string with only \\ and \$';
```

**GOOD:**
```dart
var s = r'A string with only \ and $';
```

## Related Issues

None

## Tests

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/foundation/error_reporting_test.dart [hunk 1]
```diff
@@ -70,22 +70,22 @@ Future<void> main() async {
...
-      '#0      getSampleStack\\.<anonymous closure> \\([^)]+flutter/test/foundation/error_reporting_test\\.dart:[0-9]+:[0-9]+\\)\n'
-      '#2      getSampleStack \\([^)]+flutter/test/foundation/error_reporting_test\\.dart:[0-9]+:[0-9]+\\)\n'
-      '#3      main \\([^)]+flutter/test/foundation/error_reporting_test\\.dart:[0-9]+:[0-9]+\\)\n'
-      '(.+\n)+' // TODO(ianh): when fixing #4021, also filter out frames from the test infrastructure below the first call to our main()
-      '\\(elided [0-9]+ frames from class _RawReceivePortImpl and dart:async\\)\n'
-      '\n'
-      'line 1 of extra information\n'
-      'line 2 of extra information\n'
-      '════════════════════════════════════════════════════════════════════════════════════════════════════\$',
+      r'^══╡ EXCEPTION CAUGHT BY ERROR HANDLING TEST ╞═══════════════════════════════════════════════════════\n'
```
Comment (dnfield) [inline packages/flutter/test/foundation/error_reporting_test.dart]: These `\n`s  will now be `\\n`.  I don't think we can do this one here (and below).

I also find it strange that Dart would let me concat a raw string with a non-raw string this way...

File: packages/flutter/test/foundation/error_reporting_test.dart [hunk 2]
```diff
@@ -102,16 +102,16 @@ Future<void> main() async {
...
-      'The following assertion was thrown:\n'
-      'word word word word word word word word word word word word word word word word word word word word\n'
-      'word word word word word word word word word word word word word word word word word word word word\n'
-      'word word word word word word word word word word word word word word word word word word word word\n'
-      'word word word word word word word word word word word word word word word word word word word word\n'
-      'word word word word word word word word word word word word word word word word word word word word\n'
-      '\'[^\']+flutter/test/foundation/error_reporting_test\\.dart\':\n'
-      'Failed assertion: line [0-9]+ pos [0-9]+: \'false\'\n'
-      '════════════════════════════════════════════════════════════════════════════════════════════════════\$',
+      r'^══╡ EXCEPTION CAUGHT BY FLUTTER FRAMEWORK ╞═════════════════════════════════════════════════════════\n'
```
Comment (dnfield) [inline packages/flutter/test/foundation/error_reporting_test.dart]: ditto - the `\n` at the end here is important.

File: packages/flutter/test/foundation/error_reporting_test.dart [hunk 3]
```diff
@@ -196,10 +196,10 @@ Future<void> main() async {
...
     ));
     expect(console.join('\n'), matches(
-      '^══╡ EXCEPTION CAUGHT BY FLUTTER FRAMEWORK ╞═════════════════════════════════════════════════════════\n'
-      'The following message was thrown:\n'
-      'hello\n'
-      '════════════════════════════════════════════════════════════════════════════════════════════════════\$',
+      r'^══╡ EXCEPTION CAUGHT BY FLUTTER FRAMEWORK ╞═════════════════════════════════════════════════════════\n'
+      r'The following message was thrown:\n'
+      r'hello\n'
+      r'════════════════════════════════════════════════════════════════════════════════════════════════════$',
```
Comment (dnfield) [inline packages/flutter/test/foundation/error_reporting_test.dart]: basically all of the ones in this file.  Except maybe the last line like these

**Review by dnfield — COMMENTED** [review]
This mostly seems good, and mostly seemslike a nice improvement.

I'm a bit unsure how I feel about replacing `'\\'` with `r'\'` but I guess it's fine to be consistent.

File: packages/flutter/test/foundation/error_reporting_test.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/test/foundation/error_reporting_test.dart]: > These `\n`s will now be `\\n`. I don't think we can do this one here (and below).

I think the result will be the same.

```dart
RegExp('\n').hasMatch('\n'); // true
RegExp('\\n').hasMatch('\n'); // true
```

In the first case `\n` is _expanded_ by the Dart language. In the second case it is the regex engine that handle the `\n`. Finally it seems to have the same result.

File: packages/flutter/test/foundation/error_reporting_test.dart (hunk 2 above)
Comment (a14n) [reply packages/flutter/test/foundation/error_reporting_test.dart]: not really in a regexp. See my above comment.

File: packages/flutter/test/foundation/error_reporting_test.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/test/foundation/error_reporting_test.dart]: Ahhh I missed that these are regexp.  Ok.

**Review by dnfield — APPROVED** [review]
Lgtm

**Pull request merged**